File: apps/go/routes/body/multipart.go
//go:build js && wasm

package main

import "github.com/primate-run/go/route"

var _ = route.With{ContentType: route.Multipart}.Post(func(request route.Request) any {
	multipart, err := request.Body.Multipart()
	if err != nil {
		return map[string]any{"error": err.Error()}
	}
	return multipart.Form
})

File: apps/marko/routes/route-client/multipart.ts
import response from "primate/response";
import route from "primate/route";

export default route({
  get(request) {
    return request.query.has("top-level")
      ? response.view("RouteClient/TopLevel/Multipart.marko", {
        result: JSON.stringify({ foo: "bar", file: "hello" }),
      })
      : response.view("RouteClient/Multipart.marko");
  },
  post: route.with({ contentType: "multipart/form-data" }, async request => {
    const { form, files } = await request.body.multipart();
    return {
      foo: form.foo,
      file: await files.file.text(),
    };
  }),
});
